complete.py

```diff
@@ -0,0 +1,795 @@
+            timestep : np.unit.Quantity compatible with femtoseconds, default: 1.0*unit.femtoseconds
+               Integration timestep
+            constraint_tolerance : float, default: 1.0e-6
+                Tolerance for constraint solver
+        """
+        #just super our previous method
+        super().__init__(temperature,
+                         collision_rate,
+                         timestep,
+                         splitting,
+                         constraint_tolerance,
+                         **kwargs)
+
+
+    def _add_V_step(self, force_group="0"):
+        """Deterministic velocity update, using only forces from force-group fg.
+
+        arguments
+            force_group : str, optional, default="0"
+               Force group to use for this step
+        """
+        self.addComputeSum("old_ke", self._kinetic_energy)
+
+        # update velocities
+        if self._mts:
+            self.addComputePerDof("v", "v + ((dt / {}) * modified_force / m)".format(self._force_group_nV[force_group], force_group))
+        else:
+            self.addComputePerDof("v", "v + (dt / {}) * modified_force / m".format(self._force_group_nV["0"]))
+
+        self.addConstrainVelocities()
+
+
+        self.addComputeSum("new_ke", self._kinetic_energy)
+        self.addComputeGlobal("shadow_work", "shadow_work + (new_ke - old_ke)")
+
+    def _add_F_step(self):
+        """
+        pull the mm force as a perDofVariable from the integrator to be modified
+        """
+        self.addComputePerDof('modified_force', 'f')
+
+
+    def _add_variables(self):
+        super()._add_variables()
+        self.addPerDofVariable('modified_force', 0)
+
+    def _add_integrator_steps(self):
+        """
+        Add the steps to the integrator--this can be overridden to place steps around the integration.
+        """
+        super()._add_integrator_steps()
+        #self.addUpdateContextState()
+
+    @property
+    def _step_dispatch_table(self):
+        dispatch_table = super()._step_dispatch_table
+        dispatch_table['F'] = (self._add_F_step, False) #add a modified_force variable
+        return dispatch_table
+
+
+class Propagator(OMMBIP):
+    """
+    Propagator pseudocode:
+    Step 1: initialization--
+        set iteration = 0, n_iterations = n_iterations, lambda  = 0 (i.e. iteration / n_iterations); work_accumulated = 0.0
+        generate sample x_0 ~ e^(-p(x))
+        evaluate work_incremental = 0 (i.e. u_mm(x_0) - g(x_0), but we presume that g = u_mm(.))
+        work_accumulated <- work_accumulated + work_incremental
+        x' = x_0
+    Step 2: sampling
+        for increment in range(n_iterations):
+            x = x'
+            ante_perturbation_potential =  (1 - lambda) * u_mm(x) + lambda * u_ani_mm_mix(x)
+            set iteration <- iteration + 1.0; lambda <- iteration / n_iterations
+            evaluate work_incremental = [(1 - lambda) * u_mm(x) + lambda * u_ani_mm_mix(x)] - ante_perturbation_potential
```

Comment:
ah! took me a while to realize that the difference between ante_perturbation_potential and the first part of evaluate_work_incremental changed in its value for lambda! But very nice explanation!